src/SAWScript/Proof.hs

```diff
@@ -204,6 +258,115 @@ splitProp sc (Prop p) =
             t2 <- scPiList sc vars =<< scEqTrue sc p2
             return (Just (Prop t1,Prop t2))
 
+-- | Attempt to split a disjunctive proposition into two propositions.
+splitDisj :: SharedContext -> Prop -> IO (Maybe (Prop, Prop))
+splitDisj sc (Prop p) =
+  do let (vars, body) = asPiList p
+     case (isGlobalDef "Prelude.or" <@> return <@> return) =<< asEqTrue body of
+       Nothing -> pure Nothing
+       Just (_ :*: p1 :*: p2) ->
+         do t1 <- scPiList sc vars =<< scEqTrue sc p1
+            t2 <- scPiList sc vars =<< scEqTrue sc p2
+            return (Just (Prop t1,Prop t2))
+
+-- | Attempt to split an implication into a hypothesis and a conclusion
+splitImpl :: SharedContext -> Prop -> IO (Maybe (Prop, Prop))
+splitImpl sc (Prop p)
+  | Just ( _ :*: h :*: c) <- (isGlobalDef "Prelude.implies" <@> return <@> return) =<< asEqTrue p
+  = do h' <- scEqTrue sc h
+       c' <- scEqTrue sc c
+       return (Just (Prop h', Prop c'))
+
+  -- or (not h) c == implies h c
+  | Just ( _ :*: (_ :*: h) :*: c) <- (isGlobalDef "Prelude.or" <@> (isGlobalDef "Prelude.not" <@> return) <@> return) =<< asEqTrue p
+  = do h' <- scEqTrue sc h
+       c' <- scEqTrue sc c
+       return (Just (Prop h', Prop c'))
+
+  -- or c (not h) == implies h c
+  | Just ( _ :*: c :*: (_ :*: h)) <- (isGlobalDef "Prelude.or" <@> return <@> (isGlobalDef "Prelude.not" <@> return)) =<< asEqTrue p
+  = do h' <- scEqTrue sc h
+       c' <- scEqTrue sc c
+       return (Just (Prop h', Prop c'))
+
+  -- Handle the case of (H1 -> H2), where H1 and H2 are in Prop
+  | Just (_nm, arg, c) <- asPi p
+  , looseVars c == emptyBitSet -- make sure this is a nondependent Pi (AKA arrow type)
+  = termToMaybeProp sc arg >>= \case
+      Nothing -> return Nothing
+      Just h  -> return (Just (h, Prop c))
+
+  | otherwise
+  = return Nothing
+
+
+-- | Attempt to split a sequent into two subgoals. This will only work
+--   on focused sequents. If the sequent is focused on a hypothesis,
+--   the hypothesis must be a disjunction, if/then/else, or implication term.
+--   If the sequent is focused on a conclusion, the conclusion must be
+--   a conjunction or if/then/else.
+--
+--   If this process succeeds, then a proof of the two included sequents
+--   should be sufficient to prove the input sequent.
+splitSequent :: SharedContext -> Sequent -> IO (Maybe (Sequent, Sequent))
+splitSequent sc sqt =
+  case sqt of
+    ConclFocusedSequent hs (FB gs1 g gs2) ->
```

Comment:
A very minor thing, but there are several parts of `SAWScript.Proof` that use "`gs`" to refer to conclusions. I think this is a leftover from when conclusions were called goals (hence the use of `g`)—perhaps these should be changed to "`cs`" instead?

Reply:
Ugh, yes, that happens quite a bit. I was trying to decide if it is worth it to do a fiddly variable renaming like this. What are your thoughts?

Reply:
It is indeed quite a bit of work, so I would understand if you would prefer not to do it. Can you open an issue as a reminder?